code/modules/surgery/organs/blood.dm

```diff
@@ -67,19 +67,55 @@
 			if(BP.open)
 				temp_bleed += 0.5
 
-			if(BP.internal_bleeding)
-				internal_bleeding_rate += 0.5
-
+			if(BP.arterial_bleeding && !bleedsuppress && !(HAS_TRAIT(src, TRAIT_FAKEDEATH) || src.stat >= DEAD))
+				if(world.time >= next_blood_squirt)
+					var/turf/sprayloc = get_turf(src)
+					blood_squirt(BP.arterial_bleeding_severity, sprayloc)
+					playsound(get_turf(src), 'sound/effects/artery.ogg', 75, 0)
+					visible_message("<span class='danger'>Blood squirts from [src]'s [BP.artery_name]</span>!", \
+									"<span class='userdanger'>Blood squirts from my [BP.artery_name]!</span>")
+				else
+					bleed(BP.arterial_bleeding_severity)
 		bleed_rate = max(bleed_rate - 0.5, temp_bleed)//if no wounds, other bleed effects naturally decreases
 
-		var/additional_bleed = round(clamp((reagents.get_reagent_amount("heparin") / 10), 0, 2), 1) //Heparin worsens existing bleeding
-
-		if(internal_bleeding_rate && !HAS_TRAIT(src, TRAIT_FAKEDEATH))
-			bleed_internal(internal_bleeding_rate + additional_bleed)
-
 		if(bleed_rate && !bleedsuppress && !HAS_TRAIT(src, TRAIT_FAKEDEATH))
 			bleed(bleed_rate + additional_bleed)
 
+/mob/living/carbon/human/proc/blood_squirt(amt, turf/sprayloc, spray_distance = 2)
+	if(amt <= 0 || !istype(sprayloc))
+		return
+	var/spray_dir = pick(GLOB.alldirs)
+	amt = CEILING(amt/spray_distance, 1)
+	INVOKE_ASYNC(src, .proc/do_squirt, amt, spray_dir, spray_distance)
+
+/mob/living/carbon/human/proc/do_squirt(amt, spray_dir, spray_distance)
+	bleed(amt)
+	var/turf/sprayloc = get_turf(src)
+	for(var/i = 1 to spray_distance)
+		sprayloc = get_step(sprayloc, spray_dir)
+		if(!istype(sprayloc) || sprayloc.density)
+			break
+		var/hit_mob = FALSE
+		for(var/thing in sprayloc)
+			var/atom/A = thing
+			if(!A.simulated)
+				continue
+			if(ishuman(A))
+				var/mob/living/carbon/human/H = A
+				if(!H.lying)
+					H.bloody_body(src)
+					H.bloody_hands(src)
+					to_chat(H, "<span class='danger'>You are hit by a spray of blood!</span>")
+					hit_mob = TRUE
+
+			if(hit_mob || !A.CanPass(src, sprayloc))
+				break
+
+		add_splatter_floor(sprayloc, FALSE, spraydir = spray_dir)
+		if(hit_mob)
+			return
+		sleep(1)
```

Comment:
What's the point in this `sleep` here?

Reply:
It's just to make the blood squirting do an animation instead of instantaneously splattering each tile.